spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDataSourceV2Strategy.scala

```diff
@@ -53,4 +76,18 @@ object ExtendedDataSourceV2Strategy extends Strategy {
     }
     new GenericInternalRow(values)
   }
+
+  private object IcebergCatalogAndIdentifier {
+    def unapply(identifier: Seq[String]): Option[(TableCatalog, Identifier)] = {
+      val catalogAndIdentifier = Spark3Util.catalogAndIdentifier(spark, identifier.asJava)
+      catalogAndIdentifier.catalog match {
+        case icebergCatalog: SparkCatalog =>
+          Some((icebergCatalog, catalogAndIdentifier.identifier))
+        case icebergCatalog: SparkSessionCatalog[_] =>
+          Some((icebergCatalog, catalogAndIdentifier.identifier))
```

Comment:
I don't think you need the inner paren's here

Reply:
I'm pretty sure they are needed. I've hit issues in the past with this, at least in Scala 2.11.